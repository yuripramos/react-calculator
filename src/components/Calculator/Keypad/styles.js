import styled, { css } from "styled-components";
import { rem } from "../../../styles/tools";
import { lighestgrey, trueWhite } from "../../../styles/settings";

export const KeypadWrapper = styled.div`
  text-align: center;
  color: ${trueWhite};
  margin: 0 ${rem(20)};
  padding: 0 ${rem(5)};
`;

export const Button = styled.button`
  width: ${rem(70)};
  height: ${rem(70)};
  font-size: ${rem(20)};
  background: ${lighestgrey};
`;
