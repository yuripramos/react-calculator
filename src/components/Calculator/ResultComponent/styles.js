import styled from "styled-components";
import { rem } from "../../../styles/tools";

export const Result = styled.div`
  height: ${rem(60)};
  background-color: #bbb;
  width: 100%;
`;

export const Line = styled.p`
  font-size: ${rem(40)};
  margin: ${rem(5)};
`;
