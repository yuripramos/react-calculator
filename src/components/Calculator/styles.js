import styled, { css } from "styled-components";
import { rem, media } from "../../styles/tools";

export const ContentWrapper = styled.div`
  text-align: center;
  color: #000;
  padding: ${rem(25)} ${rem(10)};
  min-height: ${rem(250)};
  width: 400px;
  margin: 0 auto;
  ${media.md(css`
    min-height: ${rem(300)};
    padding: ${rem(50)} ${rem(10)};
  `)};
`;
