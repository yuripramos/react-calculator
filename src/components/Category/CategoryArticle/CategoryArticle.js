import React, { Component  } from "react";
import moment from "moment";
import Icon from "../../common/Icon";
import { Container, Row, Column } from "../../../styles/grid";
import { string, arrayOf, shape, bool } from "prop-types";

import {
  ContentWrapper,
  Title,
  CategoryWrapper,
  PublishedDate,
  Author,
  Media,
  Clicker,
  Notice,
  SubTitle
} from "./styles";

class CategoryContent extends Component {
  constructor(props) {
    super(props);
    this.getKeyFromShortenUrl = this.getKeyFromShortenUrl.bind(this);
  }
  getKeyFromShortenUrl(hash) {
    const hashIdFull = hash.split("/");
    const shortHash = hashIdFull[hashIdFull.length - 1];
    if(shortHash.includes(".html")) {
      return shortHash.replace(".html", "");
    }
    return shortHash;
  }

  render() {
    const { content, match, isMainPage } = this.props;
    return (
      <ContentWrapper>
        <Container>
          <Row>
            <Column>
              {isMainPage && (
                <Notice>
                  <Icon name="Attention" width="50px" height="50px" />
                  This page only list the most recent news from all
                  sections, to navigate choose the sections above
                </Notice>
              )}
              {content.map((e, i) => (
                <CategoryWrapper key={`article-${e.section}-${i}`}>
                  {isMainPage && <SubTitle>in {e.section} </SubTitle>}
                  {isMainPage ? <Title>{e.title} </Title> : (
                    <Clicker
                      to={`${match.path}/${this.getKeyFromShortenUrl(
                        e.short_url || e.url
                      )}`}
                    >
                      <Title>{e.title} </Title>
                    </Clicker>
                  )}
                  <PublishedDate>
                    updated in: {moment(e.published_date).format("LLL")}
                  </PublishedDate>
                  {e.multimedia.length > 0 && (
                    <Media
                      src={
                        e.multimedia.length > 4
                          ? e.multimedia[4].url
                          : e.multimedia[2].url
                      }
                      alt="thumbnail"
                    />
                  )}
                  <Author>{e.byline} </Author>
                </CategoryWrapper>
              ))}
            </Column>
          </Row>
        </Container>
      </ContentWrapper>
    );
  }
}
CategoryContent.defaultProps = {};

CategoryContent.propTypes = {
  content: arrayOf(
    shape({
      section: string,
      abstract: string,
      url: string,
      published_date: string,
      byline: string,
      title: string
    })
  ),
  match:
    shape({
      path: string,
    }),
  isMainPage: bool,
};

export default CategoryContent;
