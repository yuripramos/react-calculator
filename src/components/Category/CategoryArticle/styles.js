import styled from "styled-components";
import { rem } from "../../../styles/tools";
import { grey10, red10, black, darkestBlue } from "../../../styles/settings";
import { Link } from "react-router-dom";

export const ContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(10)};
  min-height: ${rem(250)};
  width: 100%;
  display: inline-block;
`;

export const CategoryWrapper = styled.div`
  text-align: left;
  padding: ${rem(15)} ${rem(5)};
  min-height: ${rem(100)};
  display: inline-flex;
  flex-wrap: wrap;
  border-bottom: solid 1px ${grey10};
  width: 100%;
  :nth-child(2n) {
    background: rgba(229, 229, 232, 0.1);
  }
`;
export const Clicker = styled(Link)`
  text-decoration: none;

  &:focus,
  &:hover,
  &:visited,
  &:link,
  &:active {
    text-decoration: none;
  }
  color: ${black};
  font-size: ${rem(30)};
  font-weight: 200;
  line-height: 0.75;
  font-family: "CrimsomText SemiBold";
  margin-top: ${rem(20)};
  width: 100%;
`;

export const Media = styled.img`
  width: ${rem(400)};
  height: ${rem(300)};

`;

export const Notice = styled.div`
  font-family: "OpenSans Regular";
  font-size: ${rem(20)};
  color: ${red10};
  text-align:center;
  height: ${rem(50)};
  display: inline-flex;
  line-height: 45px;
  margin: ${rem(5)};
`;

export const Title = styled.span`
  font-size: ${rem(30)};
  font-weight: 200;
  line-height: 0.75;
  font-family: "CrimsomText SemiBold";
  margin-top: ${rem(20)};
  width: 100%;
`;

export const SubTitle = styled.span`
  margin-top: ${rem(30)};
  font-family: "OpenSans Regular";
  font-weight: normal;
  font-size: ${rem(16)};
  color: ${darkestBlue};
`;


export const PublishedDate = styled.span`
  font-family: "OpenSans Regular";
  font-size: .35em;
  margin: 1.625rem 0;
  letter-spacing: 0.05em;
  color: ${red10};
  width: 100%;
  font-size: .3em;
  margin: ${rem(20)} 0 ${rem(15)} 0;
`;

export const Author = styled.span`
  font-size: .4em;
  line-height: 0.75;
  font-family: "OpenSans Thin";
  margin: ${rem(20)} 0 ${rem(10)} 0;
  width: 100%;
`;
