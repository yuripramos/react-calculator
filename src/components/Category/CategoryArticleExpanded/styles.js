import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";
import { grey10, red10, darkestBlue } from "../../../styles/settings";

export const ContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(10)};
  min-height: ${rem(250)};
  width: 100%;
  display: inline-block;
`;

export const CategoryWrapper = styled.div`
  color: #000;
  font-family: "CrimsomText Regular";
  text-align: left;
  padding: ${rem(15)} ${rem(5)};
  min-height: ${rem(100)};
  display: inline-flex;
  flex-wrap: wrap;
  border-bottom: solid 1px ${grey10};
  :nth-child(2n) {
    background: rgba(229, 229, 232, 0.1);
  }
`;

export const Link = styled.a`
  text-decoration: none;
  font-family: "OpenSans Regular";
  font-size: ${rem(16)};
  color: ${darkestBlue};
  cursor: pointer;
`;

export const Media = styled.img`
  width: 100%;
  height: ${rem(350)};
    ${media.md(css`
    height: ${rem(500)};
  `)};
  ${media.lg(css`
    height: ${rem(620)};
  `)};
`;

export const Title = styled.span`
  font-size: ${rem(38)};
  font-weight: 200;
  line-height: 0.75;
  font-family: "CrimsomText SemiBold";
  margin-top: ${rem(30)};
  width: 100%;
`;

export const PublishedDate = styled.span`
  font-family: "OpenSans Regular";
  font-size: ${rem(16)};
  margin: 1.625rem 0;
  letter-spacing: 0.05em;
  color: ${red10};
  width: 100%;
  margin: ${rem(20)} 0 ${rem(15)} 0;
`;

export const Author = styled.span`
  font-family: "OpenSans Regular";
  font-size: ${rem(12)};
  margin: ${rem(20)} 0 ${rem(10)} 0;
  width: 100%;
`;

export const Kicker = styled.span`
  font-size: ${rem(18)};
  margin: ${rem(20)} 0 ${rem(10)} 0;
  width: 100%;
`;

export const Description = styled.span`
  font-size: ${rem(22)};
  margin: ${rem(20)} 0 ${rem(10)} 0;
  width: 100%;
`;


