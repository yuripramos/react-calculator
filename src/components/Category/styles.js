import styled, { css } from "styled-components";
import { rem, media } from "../../styles/tools";
import { trueBlack } from "../../styles/settings";

export const ContentWrapper = styled.div`
  text-align: center;
  padding: ${rem(15)} ${rem(10)};
  min-height: ${rem(250)};

  ${media.md(css`
    min-height: ${rem(300)};
    padding: ${rem(20)} ${rem(10)};
  `)};
`;
export const Title = styled.h1`
  font-family: "RobotoSlab Regular";
  font-size: ${rem(30)};
  color: ${trueBlack};
  margin-top: 0;
`;
