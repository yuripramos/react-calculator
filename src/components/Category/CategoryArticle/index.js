import CategoryArticle from "./CategoryArticle";
import { withRouter } from "react-router-dom";

export default withRouter(CategoryArticle);
