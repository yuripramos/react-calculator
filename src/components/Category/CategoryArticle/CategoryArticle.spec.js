import React from "react";
import { shallow } from "enzyme";

import CategoryArticle from "./CategoryArticle";
import { Notice, SubTitle, Clicker } from "./styles";

let props;

beforeEach(() => {
  props = {
    content: [
      {
        section: "science",
        abstract: "while ultraviolete fluor",
        url: "http://www.nytimes.com/2019/02/01/science/pink-glow",
        published_date: "2019-02-01T12:42:18-05:00",
        byline: "By VERONIQUE GREENWOOD",
        title: "Flying Squirrels That Glow Pink in the Dark",
        multimedia: [
          { url: "www.imgfor.be/23213.jpg" },
          { url: "www.imgfor.be/932sdas.jpg" },
          { url: "www.imgfor.be/3JWM3.jpg" },
          { url: "www.imgfor.be/9FSww.jpg" },
          { url: "www.imgfor.be/9FSww.jpg" }
        ]
      },
      {
        section: "science",
        abstract: "while usadasltraviolete fluor",
        url: "http://www.nytimes.com/2019/02/01/science/red-glow",
        published_date: "2019-02-01T12:42:16-05:00",
        byline: "By BARBARIAN",
        title: "Pink is the new white",
        multimedia: [
          { url: "www.imgfor.be/23213.jpg" },
          { url: "www.imgfor.be/932sdas.jpg" },
          { url: "www.imgfor.be/3JWM3.jpg" },
          { url: "www.imgfor.be/9FSww.jpg" },
          { url: "www.imgfor.be/9FSww.jpg" }
        ]
      }
    ],
    isMainPage: false,
    match: {
      path: "/science/hot"
    },

  };
});

describe("CategoryArticle Component", () => {
  it("should match snapshot with props", () => {
    expect(shallow(<CategoryArticle {...props} />)).toMatchSnapshot();
  });
  it("should warn users about functionality if they are in initial page", () => {
    const wrapper = shallow(<CategoryArticle {...props} />);
    wrapper.setProps({ isMainPage: true });
    wrapper.update();
    expect(wrapper.find(Notice).length).toBe(1);
  });
  it("should NOT warn users about functionality if they are in initial page", () => {
    const wrapper = shallow(<CategoryArticle {...props} />);
    expect(wrapper.find(Notice).length).toBe(0);
  });
  it("should render Subtitle if they are in initial page and the number be equals the lenght of the articles", () => {
    const wrapper = shallow(<CategoryArticle {...props} />);
    wrapper.setProps({ isMainPage: true });
    wrapper.update();
    expect(wrapper.find(SubTitle).length).toBe(2);
  });
  it("should NOT render Subtitle if they are in initial page", () => {
    const wrapper = shallow(<CategoryArticle {...props} />);
    expect(wrapper.find(SubTitle).length).toBe(0);
  });
  it("should NOT render Clicker link if they are in initial page and equals the length of the articles", () => {
    const wrapper = shallow(<CategoryArticle {...props} />);
    wrapper.setProps({ isMainPage: true });
    wrapper.update();
    expect(wrapper.find(Clicker).length).toBe(0);
  });
  it("should render Clicker link if they are in initial page", () => {
    const wrapper = shallow(<CategoryArticle {...props} />);
    expect(wrapper.find(Clicker).length).toBe(2);
  });
});
