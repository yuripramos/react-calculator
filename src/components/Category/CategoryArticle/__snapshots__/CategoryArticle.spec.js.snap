// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryArticle Component should match snapshot with props 1`] = `
<styled.div>
  <Container>
    <Row>
      <Column>
        <styled.div
          key="article-science-0"
        >
          <Styled(Link)
            to="/science/hot/pink-glow"
          >
            <styled.span>
              Flying Squirrels That Glow Pink in the Dark
               
            </styled.span>
          </Styled(Link)>
          <styled.span>
            updated in: 
            January 1, 2010 6:01 AM
          </styled.span>
          <styled.img
            alt="thumbnail"
            src="www.imgfor.be/9FSww.jpg"
          />
          <styled.span>
            By VERONIQUE GREENWOOD
             
          </styled.span>
        </styled.div>
        <styled.div
          key="article-science-1"
        >
          <Styled(Link)
            to="/science/hot/red-glow"
          >
            <styled.span>
              Pink is the new white
               
            </styled.span>
          </Styled(Link)>
          <styled.span>
            updated in: 
            January 1, 2010 11:01 AM
          </styled.span>
          <styled.img
            alt="thumbnail"
            src="www.imgfor.be/9FSww.jpg"
          />
          <styled.span>
            By BARBARIAN
             
          </styled.span>
        </styled.div>
      </Column>
    </Row>
  </Container>
</styled.div>
`;
